hydroflow/src/builder/surface/mod.rs

```diff
@@ -44,10 +44,17 @@ use std::hash::Hash;
 
 use crate::lang::lattice::{LatticeRepr, Merge};
 use crate::scheduled::context::Context;
+use crate::scheduled::graph::DirectedEdgeSet;
 use crate::scheduled::handoff::handoff_list::{PortList, PortListSplit};
 use crate::scheduled::port::{RECV, SEND};
 use crate::scheduled::type_list::Extend;
 
+pub trait TrackPushDependencies {
```

Comment:
Why is it we need both of these, rather than just one? The implementations I looked at looked identical between the two? Can you add a comment explaining the distinction between them?

Also, if these show up every place we require `BaseSurface`, I wonder if we could just include them in that trait. That wasn't an option when we were implementing them for the structs in the `compiled` package but might work now.

Reply:
Cool you're right -- no need to distinguish push and pull. 

Regarding `BaseSurface`, are you proposing that we move the `insert_dep` declaration into `BaseSurface` and do away with the `TrackDependencies` trait? And/or are you proposing that we can have a single `impl` for `insert_dep` that would work for any operator?

Reply:
That is what I'm proposing, I'm not entirely sure it would work without attempting it though. In the interest of merging this I propose you ignore my comment!

Reply:
I will ignore this then -- it seems kinda OK to have this as an extra trait.